tests/v2/Controller/AbstractControllerTest.php

```diff
@@ -92,10 +93,15 @@ public function assertFormIsNotValid(string $url, string $route, string $formSub
         $crawler = $client->request('GET', $url);
         $form = $crawler->selectButton(self::$translator->trans($formSubmit))->form();
         $form->setValues($values);
-        $client->submit($form);
+        $crawler = $client->submit($form);
 
         foreach ($errors as $error) {
-            $this->assertSelectorTextContains($alternateSelector ?? 'span.help-block', $this->getValidationErrorMessage($error));
+            if ($alternateSelector) {
+                $nodes = $crawler->filter($alternateSelector)->each(fn (Crawler $node) => $node->text());
+                $this->assertContains($this->getValidationErrorMessage($error), $nodes);
+            } else {
+                $this->assertSelectorTextContains($alternateSelector ?? 'span.help-block', $this->getValidationErrorMessage($error));
+            }
```

Comment:
There is something odd here, you're checking $alternateSelector twice

Reply:
oh u right, this clause is only for `'span.help-block'` selector